### ISSUE TITLE
Fix APIBuilderTests and execute them in buildbot and Travis-CI

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8518]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8518|
|Type|defect|
|Created|2016-06-27 23:36:14Z|
|Branch|https://github.com/twisted/twisted/tree/8518-apibuilder-tests|

See for example this build on trunk

https://buildbot.twistedmatrix.com/builders/debian8-py2.7/builds/500/steps/select/logs/problems

you can see that pydoctor tests are skipped 

we should have pydoctor installed using tox so that these tests are not skipped.

<details><summary>Searchable metadata</summary>

```
trac-id__8518 8518
type__defect defect
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__8518_apibuilder_tests 8518-apibuilder-tests
branch_author__ 
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__None None
time__1467070574014208 1467070574014208
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>
cc__radix
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

Tests should be fixed.
PR at https://github.com/twisted/twisted/pull/260

Please review.

The new template no longer has a link to the project main site.

I have also created a separate test case for BuildAPIDocsScript

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

Something is wrong with this branch as I see that the test is failing with

```
twisted.trial.unittest.FailTest: Project name/location not in file contents.
```

but that test was removed from the branch...

and Travis-CI is reporting the same test failure

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban commented|
|-|-|

I see what is happening here... pydoctor is installing Twisted and trial is running the installed twisted version...

will look in the morning.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

Ok. Tox is stupid but I managed to work around it.

Also, I hope that we can get rid of tox-travis as having `py27-coverage-posix,py27-nomodules-posix,topfile,narrativedocs,pyflakes` in a single log make it hard to search for skipped tests or the browse the log.

Please review and check that the tox hack is ok. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_7"></a>|@rodrigc set owner to @adiroiban|
|-|-|

extras now works on Python 3 [ticket:7807] so this patch needs to be reworked.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban removed owner|
|-|-|

I have updated the comment.

Using `.[dev]` will break the build so if we want to use extra hear we need to add a `.[trial]` here.

Also, adding extra to tox file should be a separate ticket.

In the same time, it looks like using extras in tox is broken ... see https://bitbucket.org/hpk42/tox/issues/77/cant-install-new-extras-for-target-package

Please check latest version with an update comment about using extras

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban commented|
|-|-|

Now, some tests are failing with `testlog unparseable` ... but instead of working around tox, I think that we should fix buildbot to be able to extract the test results

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl commented|
|-|-|

Buildbot is fixed, but I want to fix this in another way...

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_11"></a>|@adiroiban commented|
|-|-|

I guess that you are working in [[#8568](https://github.com/twisted/twisted/issues/8568)](https://github.com/twisted/twisted/issues/8568)

I think that this ticket is still useful to enable the build on travis-ci... as I see that #8568 is not touching that part

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_12"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset 49e548c3db53db38c28f0e09e72a8dde39e05275
```
#!CommitTicketReference repository="" revision="49e548c3db53db38c28f0e09e72a8dde39e05275"
Merge pull request #384 from twisted/docs-source-link-8568

Author: hawkowl
Reviewer: adiroiban
Fixes: #8568, #8518
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_13"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
